onnxruntime/core/providers/webnn/webnn_execution_provider.h

```diff
@@ -26,7 +26,8 @@ class WebNNExecutionProvider : public IExecutionProvider {
   GetCapability(const onnxruntime::GraphViewer& graph_viewer,
                 const IKernelLookup& /*kernel_registries*/) const override;
 
-  DataLayout GetPreferredLayout() const override { return preferred_layout_; }
+  // WebNN EP uses default NCHW layout for all backends.
+  DataLayout GetPreferredLayout() const override { return DataLayout::NCHW; }
```

Comment:
Should we add a TODO that the preferred layout needs to be queried via `context.opSupportLimits`?

WG discussion: https://github.com/webmachinelearning/webnn/issues/463#issuecomment-2132473790

Reply:
@reillyeon mentioned the current Chromium implementation focuses on more functionality than performance optimization. https://github.com/webmachinelearning/webnn/issues/463#issuecomment-2142454156

@Honry , do you happen to measure the performance before and after this PR for CPU backend?

Reply:
@huningxin, thanks much for your reminder, I measure the performance with MobileNet V2, TFLite backend with NCHW layout actually produces very poor performance.

**27.6 ms** (NCHW) vs. **6.2 ms** (NHWC)

You can also try my personal GitHub Page to compare the perf with WebNN Samples:
- NCHW: https://honry.github.io/webnn-samples/image_classification/?numRuns=101&layout=nchw
- NHWC: https://honry.github.io/webnn-samples/image_classification/?numRuns=101&layout=nhwc

cc/ @fujunwei, @reillyeon 

@fdwr, we should hold this PR until we can get better performance on TFLite backend with NCHW layout.

Reply:
I think we should expose the underlying layout preference to developers since we have the infrastructure to do that now. This would be a preference rather than a requirement but allow sites to construct a graph with the underlying device's preferred tensor layout. Note that this change shouldn't change performance (if it did that would be very surprising and require further investigation) in situations where the framework just inserts transpose operations around single operators. If the framework transposes the whole graph then it will have a benefit over the naive solution Chromium employs.

Reply:
> If the framework transposes the whole graph then it will have a benefit over the naive solution Chromium employs.

@Honry, Could you please confirm the number for MobileNet V2 ONNX model (NCHW) performance on TFLite backend through ONNXRuntime Web? 

There is a transpose optimization pass within ONNXRuntime,  https://github.com/microsoft/onnxruntime/tree/main/onnxruntime/core/optimizer/transpose_optimization, that can

> push transposes through ops and eliminate them.

Reply:
> @Honry, Could you please confirm the number for MobileNet V2 ONNX model (NCHW) performance on TFLite backend through ONNXRuntime Web?

**6.46 ms** (NHWC) vs. **27.30 ms** (NCHW)

Reply:
> 6.46 ms (NHWC) vs. 27.30 ms (NCHW)

@Honry, do you mean executing MobileNet ONNX Model (NCHW) by WebNN EP on TFLite backend is 27.30ms? Does that mean ONNXRuntime not do the whole graph transpose optimization if WebNN EP declares its preferred layout is NHWC?

Reply:
With this patch, all backends use NCHW backend in WebNN EP, so there's no graph transpose optimization from ORT applies, we get 27.30 ms on TFLite backend.

This result is similar to last one I shared about running MobileNet V2 of WebNN Samples on TFLite backend with NCHW layout: https://honry.github.io/webnn-samples/image_classification/?numRuns=101&layout=nchw, this time it depends on Chromium implementation to insert transpose op before and after every layout sensitive ops(conv, pooling etc.). MobileNet V2 has about 52 Conv nodes, that means there are 104 additional transpose ops are inserted in Chromium(maybe 52 more for transposing the weights), that's why its performance is such poor.

Reply:
I mean before this PR, what's the performance? Would the ONNX apply the whole graph transpose optimization if WebNN EP the preferred layout is NHWC for TFLite CPU backend?

Reply:
> I mean before this PR, what's the performance? Would the ONNX apply the whole graph transpose optimization if WebNN EP reports the preferred layout as NHWC for TFLite CPU backend?

It's **6.46 ms (NHWC)**, ORT does apply the whole graph tranpose optimization for NHWC preferred layout.

Reply:
> It's **6.46 ms (NHWC)**

Great to know. Created a Chromium issue for exposing preferred input layout: https://issues.chromium.org/issues/357985429

Reply:
> @fdwr, we should hold this PR until we can get better performance on TFLite backend with NCHW layout.

Acknowledged.

> 6.46 ms (NHWC) vs. 27.30 ms (NCHW)

Ooh, surprising: 4.22x = 27.3 / 6.46. A convolution is a computationally heavier operation than a transpose, and so I could see up to 3x slower, but 4x makes me curious.

> I think we should expose the underlying layout preference to developers since we have the infrastructure to do that now.

Useful.